### ISSUE TITLE
Scripting: mp.add_key_binding() doesn't work

### ISSUE THREAD

**Issue by Hydrometry** [stax76/mpv.net#271]
**Describe the bug**
This function doesn't register key binding in mpv.net, yet it does on regular mpv on Windows.

**To Reproduce**
1. put this script in AppData\Roaming\mpv.net\scripts\asdf.js
```js
mp.add_key_binding("ctrl+z", "hello", function() { mp.osd_message("hello"); });
```
2. launch mpv.net, press ctrl+z
3. doesn't work :)

Manually binding or running `script-message hello` does work however.

Cheers

**Comment by hooke007** [comment]
Because disable the built-in bindings are the default setting for mpvnet. You have to manually bind it.

**Comment by stax76** [comment]
The topic was recently raised by @avih:

https://github.com/mpv-player/mpv/issues/8809

I probably need to reread the thread because I have awful bad memory, if I remember right, then there is nothing I could do other than making a feature or pull request.

**Comment by avih** [comment]
The TL;DR is that if the (libmpv) builtin bindings are disabled (which is what mpv.net wants), then this also disables bindings added with `mp.add_key_binding` (which mpv.net doesn't want, but have to live with).

However, `mp.add_forced_key_binding` still works even when the builtin bindings are disabled.

It's a hard problem to solve, and for now there's no solution to both disable the builtin bindings and making `mp.add_key_binding` work (for keys which are not defined at `input.conf`, because if they are at input.conf then it overrides them also in normal mpv and even if the builtin bindings are not disabled).

The links posted earlier have more info, and at least the docs should be up to date.

**Comment by avih** [comment]
Also, I tried (with the new dll): `mpv.net --enable-builtin-bindings=yes` but got an error:
```
[mpv.net] error setting property: enable-builtin-bindings = yes
[mpv.net] property not found
```

However, you don't seem to remove unknown properties at your `mpv.conf` - which is how I was able to test `input-builtin-bindings`.

I _think_ it could be useful to accept unknown option `--ANYTHING=WHATEVER` form and simply pass it on to mpv, unless you have a reason to curate the list of command line options (but not config files?) to pass on to mpv?

**Comment by stax76** [comment]
OK, thanks. I'm now subscribed to the pull request thread.

I changed it using shinchiro built from today.

> To avoid similar issues in the future, maybe you shouldn't write mpv.net defaults into the config files? I.e. apply them, and allow the config to override them, but if you modify the defaults and the user never overrides them in mpv.conf, then they'll just have the new defaults without issues.

That is probably a better solution, thanks, I've rewritten it accordingly.

For `input-default-bindings` it's now difficult finding a good solution because it's already in existing conf files. I decided to enforce `input-default-bindings=yes` and `input-builtin-bindings=no` in the source code after init, that solves the backward compatibility problem and hopefully does not create other problems, I don't think it can make sense for mpv.net users to use mpv defaults.

**Comment by avih** [comment]
> I decided to enforce input-default-bindings=yes and input-builtin-bindings=no in the source code after init,

input-builtin-bindings=no after init is ignored. Its value is only applied during init.

forcing input-default-bindings=yes is not a great idea IMO (and same for input-builtin-bindings=no, even if it currently doesn't have any effect if you do that after init). What if the user wants their mpv kb shortcuts?

I also thought about it and I agree it's not a trivial problem to solve.

One solution I thought could work for this case is to simply remove it from the user mpv.conf but only once (or replace it with some comment), and next time remember that you already removed it - and don't remove it again if the user put it there again.

For instance you can store some "removed-default-bindings-once=yes" at settings.xml, and next time, if it exists then don't remove it again?

**Comment by avih** [comment]
https://github.com/stax76/mpv.net/commit/6e761c0a4e02be191ad4199b604a3170f3a7ced5#diff-c8482043a97be5b4aced3348ffa18cfc898a44b1691682bf32f5d4d0c1f6ff6aL151

This completely removes the mpv.net KB delay/repeat value, right? If yes, this is a good change. I was wondering why you want to override the user's preferred/system KB timing settings... :)

**Comment by hooke007** [comment]
https://github.com/stax76/mpv.net/pull/272/files#diff-fd29fc106cb149d9950ead25efde66a80addbc4f55491aacac023ea85be31895R121

Not me. but no matter. 10 could be rapid for some users.

**Comment by stax76** [comment]
Oh sorry, wasn't you. I don't remember exactly why it wasn't working with my old MCE remote, maybe it did not auto repeat.
